src/internals/hooks/useEventCallback.ts

```diff
@@ -1,12 +1,18 @@
-import { useEffect, useCallback, useRef } from 'react';
+import { useCallback, useRef } from 'react';
+import useIsomorphicLayoutEffect from './useIsomorphicLayoutEffect';
 
 /**
  * https://github.com/facebook/react/issues/14099#issuecomment-440013892
  * @param {function} fn
  */
 export function useEventCallback(fn) {
   const ref = useRef(fn);
-  useEffect(() => {
+  /**
+   * use useLayoutEffect instead of useEffect.
+   * useLayoutEffect is earlier than useEffect, sometimes we use setState and then use callback immediately,
+   * However the state in callback is not the latest, because useEffect is not triggered.
+   */
+  useIsomorphicLayoutEffect(() => {
```

Comment:
The content of this file is not within the scope of this PR.

Reply:
Sorry about this, I don't know what happened here. I'm removing this from the PR